runtime/compiler/aarch64/codegen/J9ARM64Snippet.cpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "j9.h"
+
+#ifdef J9VM_TASUKI_LOCKS_SINGLE_SLOT
+
+#include "codegen/CodeGenerator.hpp"
+#include "codegen/J9ARM64Snippet.hpp"
+
+TR::ARM64MonitorEnterSnippet::ARM64MonitorEnterSnippet(
+   TR::CodeGenerator *codeGen,
+   TR::Node *monitorNode,
+   int32_t lwOffset,
+   TR::LabelSymbol *incLabel,
+   TR::LabelSymbol *callLabel,
+   TR::LabelSymbol *restartLabel)
+   : _incLabel(incLabel),
+     _lwOffset(lwOffset),
+     TR::ARM64HelperCallSnippet(codeGen, monitorNode, callLabel, monitorNode->getSymbolReference(), restartLabel)
+   {
+   TR_ASSERT(false, "Not implemented yet");
```

Comment:
General comment.  Please use the new `TR_UNIMPLEMENTED()` macro for unimplemented functions throughout this PR.